### PR TITLE
Feature/1153 refactor plant layer api

### PR DESCRIPTION
**Pull request by Bushuo**
## Basics

- [ ] I added a line to [changelog.md](/doc/changelog.md)
- [ ] The PR is rebased with current master.
- [ ] Details of what you changed are in commit messages.
- [ ] References to issues, e.g. `close #X`, are in the commit messages and changelog.
- [ ] The buildserver is happy.

## Checklist

- [ ] I have installed and I am using [pre-commit hooks](../doc/contrib/README.md#Hooks)
- [ ] I fully described what my PR does in the documentation
- [ ] I fixed all affected documentation
- [ ] I fixed the introduction tour
- [ ] I wrote migrations in a way that they are compatible with already present data
- [ ] I fixed all affected decisions
- [ ] I added automated tests or a [manual test protocol](../doc/tests/manual/protocol.md)
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](https://github.com/ElektraInitiative/PermaplanT/blob/master/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)
- [ ] Exceptions to any guidelines are documented

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] I've read through the whole documentation
- [ ] I've checked conformity to guidelines

### CODE REVIEW

**Review by horenso — APPROVED** [review]

**Review by horenso — COMMENTED** [review]
Looks good :) I'm rebasing it on master since feature/1067-store-planting-note-in-backend changed a few things but shouldn't be difficult.

File: backend/src/controller/plantings.rs [hunk 1]
```diff
@@ -107,57 +94,51 @@ pub async fn create(
     params(
         ("map_id" = i32, Path, description = "The id of the map the layer is on"),
     ),
-    request_body = UpdatePlantingDto,
+    request_body = ActionDtoWrapperUpdatePlantings,
```
Comment (markus2330) [inline backend/src/controller/plantings.rs:97]: Do we really want so long names? Why no simply UpdatePlantingsDto? (plural for collection)

File: backend/src/model/dto/actions.rs [hunk 2]
```diff
@@ -23,31 +29,40 @@ use super::{BaseLayerImageDto, UpdateMapGeometryDto};
...
     /// An action used to broadcast an update to the map gemetry.
     UpdateMapGeometry(UpdateMapGeometryActionPayload),
     /// An action used to update the `additional_name` of a plant.
     UpdatePlantingAdditionalName(UpdatePlantingAdditionalNamePayload),
 }
 
+#[typeshare]
+#[derive(Debug, Serialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct ActionWrapper<T> {
```
Comment (markus2330) [inline backend/src/model/dto/actions.rs:60]: Please shortly describe all structs what they are for etc.

File: backend/src/model/entity/plantings.rs [hunk 3]
```diff
@@ -56,6 +56,9 @@ pub struct Planting {
 #[derive(Debug, Clone, Default, AsChangeset)]
 #[diesel(table_name = plantings)]
 pub struct UpdatePlanting {
+    /// The id of the planting.
+    /// This is not updated.
```
Comment (markus2330) [inline backend/src/model/entity/plantings.rs:60]: When should it be updated?

**Review by markus2330 — APPROVED** [review]
Great job. Is the problem with the transaction already fixed? How?

**Review by horenso — APPROVED** [review]

**Pull request merged**